src/pubsub/ua_pubsub_manager.c

```diff
@@ -719,21 +719,30 @@ UA_PubSubManager_generateUniqueGuid(UA_Server *server) {
     }
 }
 
-/* Initialization the PubSub configuration. */
-void
-UA_PubSubManager_init(UA_Server *server, UA_PubSubManager *pubSubManager) {
-    /* ToDo: Must be correctly generated from Mac address and port. */
+static UA_UInt64
+generateRandomUInt64(UA_Server *server) {
+    UA_UInt64 id = 0;
     UA_Guid *ident = NULL;
     ident = UA_Guid_new();
     if(!ident) {
         UA_LOG_ERROR(&server->config.logger, UA_LOGCATEGORY_SERVER,
                      "PubSub generate defaultPublisherId failed.");
-        return;
+        return 0;
     }
     *ident = UA_Guid_random();
```

Comment:
A small nitpick that was already in the code:
UA_Guid_random returns a scalar.
So you can stack-allocate `ident` without doing the malloc/free dance.